src/instructlab/model/evaluate.py

```diff
@@ -0,0 +1,504 @@
+
+        print("Evaluating answers...")
+        proc = None
+        try:
+            proc = subprocess.Popen(
+                [
+                    "python",
+                    "-m",
+                    "vllm.entrypoints.openai.api_server",
+                    "--model",
+                    judge_model,
+                    "--tensor-parallel-size",
+                    "1",
+                    "--served-model-name",
+                    "judge_model",
+                ]
+            )
+            time.sleep(60)
+            overall_score, qa_pairs, turn_scores = evaluator.judge_answers(
+                "http://localhost:8000/v1"
+            )
+            print("# SKILL EVALUATION REPORT")
+            display_model(model)
+            print("\n### AVERAGE:")
+            print(f"{round(overall_score, 2)} (across {len(qa_pairs)})")
+            print("\n### TURN ONE:")
+            print(round(turn_scores[0], 2))
+            print("\n### TURN TWO:")
+            turn2_score = turn_scores[1]
+            if isinstance(turn2_score, float):
+                turn2_score = round(turn2_score, 2)
+            print(turn2_score)
+        finally:
+            if proc:
+                proc.terminate()
+
+    elif benchmark == "mt_bench_branch":
+        # TODO: Replace temp Popen hack with serving library calls.  Current library doesn't support server-model-name.
+        evaluators = [
+            evaluator,
+            MTBenchBranchEvaluator(
+                "base_test_model",
+                "base_judge_model",
+                taxonomy_path,
+                base_branch,
+                output_dir,
+                max_workers,
+            ),
+        ]
+        branches = [branch, base_branch]
+        m_paths = [model, base_model]
+        m_names = ["test_model", "base_test_model"]
+        judge_model_names = ["judge_model", "base_judge_model"]
+        qa_pairs_list = []
+
+        for i, evaluator in enumerate(evaluators):
+            branch = branches[i]
+            m_path = m_paths[i]
+            m_name = m_names[i]
+            j_model_name = judge_model_names[i]
+
+            print(
+                f"Generating questions and reference answers from qna files for branch {branch}..."
+            )
+            try:
+                proc = subprocess.Popen(
+                    [
+                        "python",
+                        "-m",
+                        "vllm.entrypoints.openai.api_server",
+                        "--model",
+                        m_path,
+                        "--tensor-parallel-size",
+                        "1",
+                        "--served-model-name",
+                        m_name,
+                    ]
+                )
+                time.sleep(60)
+                evaluator.gen_answers("http://localhost:8000/v1")
+            finally:
+                proc.terminate()
+
+            print(f"Evaluating answers for branch {branch}...")
+            try:
```

Comment:
This server launch could be shared between the two models.  Addressed here: https://github.com/instructlab/instructlab/compare/main...danmcp:instructlab:evalandvllm?expand=1